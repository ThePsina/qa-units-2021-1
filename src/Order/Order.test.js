import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {fakeOrders} from "../data/fakeOrders";
import Order from './Order'

jest.mock('../utils/getDate');
/* eslint-disable import/first */
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter()});

describe('Order.js', () => {
    beforeEach(() => {
        getDate.mockReturnValue('hello from GoLang');
    })

    afterEach(() => {
        jest.resetModules()
    })

    it('right render', () => {
        const wrapper = shallow(<Order
            order={{shop: 'shop', date: 1234567890}}
        />);

        expect(wrapper).toMatchSnapshot();
    });

    it('render without order', () => {
        const wrapper = shallow(<Order/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('render without items', () => {
        const wrapper = shallow(<Order order={
            {
                id: 228,
                date: 1234567891,
                shop: 'Alihandro Express',
            }
        }/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render without shop', () => {
        const wrapper = shallow(<Order order={
            {
                id: 69,
                date: 1234567891,
                items: ['item1', 'item2']
            }
        }/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('all in input', () => {
        const wrapper = shallow(<Order order={
            {
                shop: 'shop',
                id: 69,
                date: 1234567891,
                items: ['item1', 'item2']
            }
        }/>);
        expect(wrapper).toMatchSnapshot();
    });
});
