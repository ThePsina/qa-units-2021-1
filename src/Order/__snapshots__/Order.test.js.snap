// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js all in input 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      hello from GoLang
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      item1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      item2
    </div>
  </div>
</div>
`;

exports[`Order.js render without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      hello from GoLang
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js right render 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      shop
    </span>
    <span
      id="date"
    >
      hello from GoLang
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;
